File: DatingApp/API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams
    {
        private const int MaxPageSize = 100;

        public int PageNumber { get; set; } = 1;

        private int pageSize = 18;

        public int PageSize
        {
            get => pageSize;
            set => pageSize = value > MaxPageSize ? MaxPageSize : value;
        }

        public string CurrentUsername { get; set; }

        public string Gender { get; set; }

        public int MinAge { get; set; } = 18;

        public int MaxAge { get; set; } = 100;
    }
}

File: DatingApp/client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 100;
    pageNumber = 1;
    pageSize = 12;
    orderBy = 'lastActive';
    constructor(user: User) {
        this.gender = user.gender === 'female' ? 'male' : 'female'
    }
}